src/main/scala/nl/knaw/dans/easy/stage/dataset/AdditionalLicense.scala

```diff
@@ -44,18 +45,40 @@ object AdditionalLicense {
     } yield mime
 
   def getAdditionalLicenseTemplate(ddm: NodeSeq)(implicit s: Settings): Try[(String, MimeType)] = Try {
+
     val licenses = ddm \\ "DDM" \ "dcmiMetadata" \ "license"
     licenses match {
       case Seq(license) =>
           if(hasXsiType(license, "http://purl.org/dc/terms/", "URI")) {
-            val licenseTemplateFile = s.licenses(license.text)
+            val licenseTemplateFile = getMatchingLicense(new URI(license.text), s.licenses).getOrElse(throw RejectedDepositException(s"Not a valid license URI: ${license.text}"))
             (FileUtils.readFileToString(licenseTemplateFile, "UTF-8"), getLicenseMimeType(licenseTemplateFile.getName))
           }
           else (license.text, "text/plain")
       case lics => throw RejectedDepositException(s"Found ${lics.size} dcterms:license elements. There should be exactly one")
     }
   }
 
+  /**
+   * Retrieves a matching license, while being liberal in what it excepts:
+   *
+   * - https is excepted instead of http
+   * - a trailing slash is ignored
+   *
+   * @param uri the URI of the license
+   * @param licenses the map from license URI-string to license File
+   * @return
```

Comment:
specify what it returns:
* when does the user receive a `Some`
* when does the user receive a `None`
* is the file relative or absolute